curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb583dadb33a77595797bd.md

```diff
@@ -0,0 +1,423 @@
+---
+id: 64cb583dadb33a77595797bd
+title: Step 106
+challengeType: 0
+dashedName: step-106
+---
+
+# --description--
+
+Add another conditional statement that checks if the player's `position.y` is greater than or equal to the checkpoint's `position.y`.
+
+Below that statement, add another conditional statement that checks if the player's `position.y` plus the player's `height` is less than or equal to the checkpoint's `position.y` plus the checkpoint's `height`.
+
+For the last array item, add the `isCheckpointCollisionDetectionActive` variable.
+
+# --hints--
+
+TODO: add tests
+
+```js
+
+```
```

Comment:
These work:

You should have a conditional statement that checks if the player's `position.y` is greater than or equal to the checkpoint's `position.y`

```js
assert.match(code, /player\.position\.y\s*>=\s*checkpoint\.position\.y,/)
```

You should have a conditional statement that checks if the player's `position.y` plus the player's `height` is less than or equal to the checkpoint's `position.y` plus the checkpoint's `height`.

```js
assert.match(code, /player\.position\.y\s*>=\s*checkpoint\.position\.y,\s*player\.position\.y\s*\+\s*player\.height\s*<=\s*checkpoint\.position\.y\s*\+\s*checkpoint\.height,/)
```

You should add `isCheckpointCollisionDetectionActive` as the last item of the `checkpointDetectionRules` array.

```js
assert.match(code, /player\.position\.y\s*>=\s*checkpoint\.position\.y,\s*player\.position\.y\s*\+\s*player\.height\s*<=\s*checkpoint\.position\.y\s*\+\s*checkpoint\.height,\s*isCheckpointCollisionDetectionActive\s*/)
```